selene_sdk/utils/config_utils.py

```diff
@@ -205,79 +207,48 @@ def create_data_source(configs, output_dir=None, load_train_val=True, load_test=
+                task_config["transform"] = transform
+            task_dataset = dataset_class(**task_config)
+
+            # create sampler
+            sampler = None
+            if task+"_sampler_class" in dataset_info:
+                sampler_class = getattr(module, dataset_info[task+"_sampler_class"])
+                if task+"_sampler_args" not in dataset_info:
+                    sampler_args = {}
                 else:
-                    val_sampler_args = dataset_info["validation_sampler_args"]
-                if not "generator" in val_sampler_args:
-                    val_sampler_args["generator"] = gen
-                val_sampler = val_sampler_class(val_dataset, **val_sampler_args)
-
-            val_loader = torch.utils.data.DataLoader(
-                val_dataset,
-                batch_size=dataset_info["loader_args"]["batch_size"],
-                num_workers=dataset_info["loader_args"]["num_workers"],
-                worker_init_fn=module.encode_worker_init_fn,
-                sampler=val_sampler
-            )
-            if not load_test:
-                return train_loader, val_loader
+                    sampler_args = dataset_info[task+"_sampler_args"]
+                if not "generator" in sampler_args:
+                    gen = torch.Generator()
+                    gen.manual_seed(configs["random_seed"])
+                    sampler_args["generator"] = gen
+                sampler = sampler_class(task_dataset, **sampler_args)
+
+            task_loader = torch.utils.data.DataLoader(
+                    task_dataset,
+                    batch_size=dataset_info["loader_args"]["batch_size"],
+                    num_workers=dataset_info["loader_args"]["num_workers"],
+                    worker_init_fn=module.encode_worker_init_fn,
+                    sampler=sampler,
+                )
+            loaders[task] = task_loader
+
+        if load_train_val and not load_test:
+                return loaders["train"], loaders["validation"]
         if load_test:
-            # load test dataset and loader
-            test_config = dataset_info["dataset_args"].copy()
-            test_config["intervals"] = test_intervals
-            if "test_transform" in dataset_info:
-                # load transforms
-                test_transform = instantiate(dataset_info["test_transform"])
-                test_config["transform"] = test_transform
-            test_dataset = dataset_class(**test_config)
-
-            test_loader = torch.utils.data.DataLoader(
-                test_dataset,
-                batch_size=dataset_info["loader_args"]["batch_size"],
-                num_workers=dataset_info["loader_args"]["num_workers"],
-                worker_init_fn=module.encode_worker_init_fn,
-            )
             if not load_train_val:
-                return test_loader
-        return train_loader, val_loader, test_loader
+                return loaders["test"]
```

Comment:
Sorry, GH is not letting me make a suggestion to delete these selected lines. The point is to delete these lines and accept all the other suggestions, so that you only load data sources specified as arguments